File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int, int> count;
        int result = 0;
        int maxCount = 0;

        for (int num : nums) {
            count[num]++;
            if (count[num] > maxCount) {
                result = num;
                maxCount = count[num];
            }
        }
        return result;
    }
};

File: 169-majority-element/majority-element.rs
impl Solution {
    pub fn majority_element(nums: Vec<i32>) -> i32 {
        let mut count = 0;
        let mut dup = 0;
        for num in nums {
            if count == 0 {
                dup = num;
            }
            if num == dup {
                count += 1;
            } else {
                count -= 1;
            }
        }
        dup
    }
}